### PR TITLE
Improve actions

### PR DESCRIPTION
**Pull request by jesseleite**
- Adds `visibleToBulk()` method to better control when bulk actions are shown.
- Implements @jackmcdade's suggestion (see https://github.com/statamic/cms/pull/1650#issuecomment-611816554) on publish/unpublish actions.
- Removes `$bulk` property.
    - Introduced by https://github.com/statamic/cms/pull/1612; Can use `visibleToBulk()` method instead.
- Renames `filter()` to `visibleTo()`.
    - For consistency with `visibleToBulk()` and our `visibleTo()` pattern.

### CODE REVIEW

File: resources/js/components/data-list/BulkActions.vue [hunk 1]
```diff
@@ -27,7 +27,8 @@
 </template>
 
 <script>
-import Actions from './Actions.vue';
+import Actions from './Actions.js';
```
Comment (jasonvarga) [inline resources/js/components/data-list/BulkActions.vue:30]: You don't need `.js` when it's a js file. `'./Actions'` will do.

File: resources/js/components/data-list/BulkActions.vue [hunk 2]
```diff
@@ -42,56 +43,52 @@ export default {
...
-            // side authorization for when the action is executed anyway, just in case.
-            rows.forEach(row => {
-                actions = actions.filter(action => row.actions.map(a => a.handle).includes(action.handle));
-            });
-
-            return _.sortBy(actions, 'title');
-        }
-
+    watch: {
+        selections: 'getActions',
```
Comment (jasonvarga) [inline resources/js/components/data-list/BulkActions.vue:63]: I didn't know you could do this. Neat!

File: resources/js/components/data-list/InlineActions.vue [hunk 3]
```diff
@@ -21,7 +21,7 @@
 
 
 <script>
-import Actions from './Actions.vue';
+import Actions from './Actions.js';
```
Comment (jasonvarga) [inline resources/js/components/data-list/InlineActions.vue:24]: .js here too

File: routes/cp.php [hunk 4]
```diff
@@ -41,7 +41,8 @@
 
         Route::group(['prefix' => 'collections/{collection}/entries'], function () {
             Route::get('/', 'EntriesController@index')->name('collections.entries.index');
-            Route::post('actions', 'EntryActionController')->name('collections.entries.actions');
+            Route::post('actions', 'EntryActionController@run')->name('collections.entries.actions.run');
+            Route::get('actions', 'EntryActionController@bulkActions')->name('collections.entries.actions.bulk-actions');
```
Comment (jasonvarga) [inline routes/cp.php:45]: Let's call the routes `.bulk` instead of `.bulk-actions`

File: src/Actions/Action.php [hunk 5]
```diff
@@ -20,13 +20,35 @@ abstract class Action implements Arrayable
...
     protected $context = [];
-    protected $bulk = true;
 
-    public function filter($item)
+    public function visibleTo($item)
     {
         return true;
     }
 
+    public function visibleBulk($items)
```
Comment (jasonvarga) [inline src/Actions/Action.php:29]: If we're renaming `visibleTo`, should we call this `visibleToBulk`?

File: resources/js/components/data-list/BulkActions.vue (hunk 2 above)
Comment (jesseleite) [reply resources/js/components/data-list/BulkActions.vue:63]: Yes, we're [doing it here already](https://github.com/statamic/cms/blob/master/resources/js/components/data-list/FieldFilter.vue#L127) 👍

**Pull request merged**